warehouse/legacy/api/simple.py

```diff
@@ -74,14 +73,6 @@ def simple_detail(project, request):
     # Get the latest serial number for this project.
     request.response.headers["X-PyPI-Last-Serial"] = str(project.last_serial)
 
-    # Get all of the files for this project.
-    files = sorted(
-        request.db.query(File)
-        .options(joinedload(File.release))
-        .join(Release)
-        .filter(Release.project == project)
-        .all(),
-        key=lambda f: (parse(f.release.version), f.filename),
-    )
+    render_simple_detail(project, request, store=True)
```

Comment:
This is only inline here at the moment to actually trigger a render/store.

Reply:
Makes sense! It looks like the current implementation unconditionally pushes to the object store when `store=True` on every request; is that intended (vs. checking whether the stored version is stale first)?

Reply:
My assumption is that `render_simple_detail` would be called primarily in response to uploads, thus would only be retouched if a task retries. So with that in mind and given the fact that these are immutable and for all intents and purposes impossible to collide, it seemed like it wouldn't be a concern.

Is there something lurking I need to be concerned with 😅. Given that we *do* generate these primarily based on user supplied inputs... I guess a very clever and lucky attacker could force an overwrite of an existing index page?

Reply:
> Given that we _do_ generate these primarily based on user supplied inputs... I guess a very clever and lucky attacker could force an overwrite of an existing index page?

I _think_ this is safe, at least assuming that an attacker can't cause a conflict with `project.normalized_name` and that there isn't a feasible preimage attack on BLAKE2b. I think both of those assumptions hold, so it should be fine 🙂